include/gunrock/applications/async/queue.cuh

```diff
@@ -0,0 +1,339 @@
+#ifndef QUEUE
+#define QUEUE
+
+#include <inttypes.h>
+#include <limits>
+#include <assert.h>
+
+#include "./util/error_util.cuh"
+#include "./util/util.cuh"
+
+namespace MaxCountQueue {
+#define PADDING_SIZE (64)  // 32 gets bad performance
+
+template<typename T, typename CounterT>
+struct Queue {
```

Comment:
This is actually really cool to see, we can safely assume the Queue data structure is an integral part of the implementation. It's arguably more important than writing the asynchronous algorithm itself. I had this idea initially where all of these unique datasets can be abstracted away nicely, and be exposed in `gunrock/container/`, from where, async could be _one_ of the targetted audience for the queue container.